todo_app/templates/index.html

```diff
@@ -2,17 +2,70 @@
 {% block title %}To-Do App{% endblock %}
 
 {% block content %}
-  <div class="jumbotron">
-    <h1 class="display-4">To-Do App</h1>
-    <p class="lead">Just another to-do app.</p>
-  </div>
+	<div class="jumbotron">
+		<h1 class="display-4">To-Do App</h1>
+		<p class="lead">Just another to-do app.</p>
+	</div>
+	<div class="row">
+		<div class="col-md-12">
+			<form method="post" action="/todo/add">
+				<div class="mb-3">
+					<div class="input-group mb-3">
+						<input 
+							type="text" 
+							class="form-control" 
+							id="new-todo-item" 
+							name="new-todo-item" 
+							placeholder="Get stuff done!"
+						/>
+						<div class="input-group-append">
+							<button class="btn btn-primary" type="submit">Add Todo</button>
+						</div>
+					</div>
+				</div>
+			</form>
 
-  <div class="row">
-    <div class="col-md-12">
-      <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
-    </div>
-  </div>
+			<h2>
+				Items
+				<span class="badge badge-primary">
+					{{ num_todos }}
+				</span>
+			</h2>
+			<p>
+				{% if num_todos == 0 %}
+					Nothing here yet!
+				{% elif num_complete_todos == num_todos %}
+					Everything is complete, you can relax for now!
+				{% else %}
+					What are you waiting for, there's {{ num_todos - num_complete_todos }} left!
+				{% endif %} 
```

Comment:
Nice 😄 We'll talk in a later module about ViewModels which can be nice ways to capture logic so that we don't have to have quite as much logic in our HTML as this can be quite hard to maintain, e.g. maybe we could capture these half a dozen lines later with:
```
{{ view_model.get_todos_status_message() }}
```
And have the logic on what gets displayed captured in our Python